engine/script/src/script_sys.cpp

```diff
@@ -177,7 +178,14 @@ union SaveLoadBuffer
         if (!result)
         {
             dmSys::Unlink(tmp_filename);
+
+        #if defined(DM_NO_ERRNO)
             return luaL_error(L, "Could not write to the file %s.", filename);
+        #else
+            char errmsg[128] = {};
+            dmStrError(errmsg, sizeof(errmsg), errno);
+            return luaL_error(L, "Could not write to the file %s, reason: %s.", filename, errmsg);
```

Comment:
According to unix and windows manuals, fwrite _should_ set errno when it fails, so this should be enough..